test/extended/router/gatewayapicontroller.go

```diff
@@ -231,31 +206,50 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 	})
 
 	g.It("Ensure HTTPRoute object is created", func() {
+		// skip since gateway needs LB service
+		skipGatewayIfNonCloudPlatform(oc)
+
 		g.By("Ensure default GatewayClass is accepted")
 		errCheck := checkGatewayClass(oc, gatewayClassName)
 		o.Expect(errCheck).NotTo(o.HaveOccurred(), "GatewayClass %q was not installed and accepted", gatewayClassName)
 
 		g.By("Getting the default domain")
 		defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
-		o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to find default domain name")
 		customDomain := strings.Replace(defaultIngressDomain, "apps.", "gw-custom.", 1)
 
 		g.By("Create a custom Gateway for the HTTPRoute")
 		gw := names.SimpleNameGenerator.GenerateName("gateway-")
 		gateways = append(gateways, gw)
 		_, gwerr := createAndCheckGateway(oc, gw, gatewayClassName, customDomain)
-		o.Expect(gwerr).NotTo(o.HaveOccurred(), "failed to create Gateway")
+		o.Expect(gwerr).NotTo(o.HaveOccurred(), "Failed to create Gateway")
 
 		g.By("Create the http route using the custom gateway")
-		httproutes = append(httproutes, "test-httproute")
-		defaultRoutename := "test-hostname.gwapi." + customDomain
+		defaultRoutename := "test-hostname." + customDomain
 		createHttpRoute(oc, gw, "test-httproute", defaultRoutename, "echo-pod-"+gw)
 
 		g.By("Checking the http route using the default gateway is accepted")
 		assertHttpRouteSuccessful(oc, gw, "test-httproute")
 	})
 })
 
+func skipGatewayIfNonCloudPlatform(oc *exutil.CLI) {
+	infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(infra).NotTo(o.BeNil())
+
+	platformType := infra.Status.Platform
+	if infra.Status.PlatformStatus != nil {
+		platformType = infra.Status.PlatformStatus.Type
+	}
```

Comment:
Just noticed that we set the platform type with the deprecated field:
```
    // platform is the underlying infrastructure provider for the cluster.
    //   
    // Deprecated: Use platformStatus.type instead.
    Platform PlatformType `json:"platform,omitempty"`

    // platformStatus holds status information specific to the underlying
    // infrastructure provider.
    // +optional
    PlatformStatus *PlatformStatus `json:"platformStatus,omitempty"`
```
I suppose that if `infra.Status.PlatformStatus==nil`  then `infra.Status.Platform` can be of no use too (e.g. empty).

We can set `platformType := ""` and add `o.Expect()` to check that we got the platform type from `infra.Status.PlatformStatus.Type`.

Reply:
Change as below
```
platformType := infra.Status.PlatformStatus.Type
o.Expect(platformType).NotTo(o.BeNil())
```